metabase/metabase
# fix(sdk): static question should cancel requests on component unmount

### Description

When the `StaticQuestion` component unmounts, e.g. when we navigate to other pages, the old requests are still running. This PR adds request cancellation to the StaticQuestion component.

### How to verify

- Open Shoppy
- Use the switch shop button to switch shop
- You should see requests cancelled per the below screenshot

### Demo

Pending requests are getting properly cancelled. Notice the **(canceled)** in DevTools.

![CleanShot 2567-10-17 at 00 18 52@2x](https://github.com/user-attachments/assets/a4d3ba80-1173-4246-b280-e586a468a669)
